### ISSUE TITLE
Files sent to computer with blueman end up in wrong folder

### ISSUE THREAD

**Issue by insomniacno1** [blueman-project/blueman#1069]
blueman: 2.0.4
BlueZ: bluez (5.37-0ubuntu5.1) xenial-security; urgency=medium
Distribution: Peach OSI 16.04 LTS 64bit (Ubuntu 16.04.6 LTS) Xubuntu
Kernel: Linux 4.4.0-154-generic
Desktop environment: XFCE 4 Gnome
Creator: https://www.peachosi.com/ and can be contacted by sending a message here https://www.facebook.com/peachosi

* [x] I have consulted the [Troubleshooting page](https://github.com/blueman-project/blueman/wiki/Troubleshooting) and done my best effort to follow.

My Issue is not in the Troubleshooting page, though it has been reported before but not resolved. https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=836982

Everything works fine, except for the placement of files sent to the computer with blueman. I have set the placement folder in the applet and in blueman-manager to Downloads. Nonetheless the files ends up in /home/username/.cache/obexd/ 

Then I found this solution online:
-----------------------------------
When Bluetooh get screwed up and incomming files go to /home/username/.cache/obexd/ instead of going to /home/username/Downloads/ then try this from terminal:

sudo apt-get update && sudo apt-get install gnome-user-share

Once that is done

gnome-file-share-properties

See if that is set to accept files over bluetooth to Downloads, then log out and in again. Now files should go to Downloads instead.

More here: https://2buntu.com/articles/1410/receiving-files-via-bluetooth-on-ubuntu/
-----------------

Without logging out or restart files now go to Downloads folder. After restart the files go to Downloads folder. 

BUT after installing updates of any kind, the files go to /home/username/.cache/obexd/

Even if I make a new folder or chose something else than Downloads, then it happens again after update - its like blueman gets reset to factory default by update, even if its not blueman that get updated.

This is really annoying, and I'm not sure what is causing the problem, but I'm willing to help you figure it out, so just let me know how I can help. I can live with it, but it is kinda stupid because in both blueman applet and bluman-manager it stil is set to Downloads, but apparently ignored, even though these are loaded on startup. 

Debug log:
```
blueman-applet --loglevel debug
blueman-applet version 2.0.4 starting
Stale PID, overwriting
_________
Load (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:60)
['StandardItems', 'PPPSupport', 'ShowConnected', 'NetUsage', 'Menu', 'RecentConns', 'KillSwitch', 'Networking', 'AuthAgent', 'ExitItem', 'NMDUNSupport', 'DBusService', 'PowerManager', 'StatusIcon', 'SerialManager', 'AppIndicator', 'DhcpClient', 'DiscvManager', 'NMPANSupport', 'Headset', 'TransferService', 'GameControllerWakelock'] 
_________
get_interface_version (/usr/lib/python3/dist-packages/blueman/bluez/BlueZInterface.py:13)
Detected BlueZ 5 
/usr/lib/python3/dist-packages/blueman/plugins/applet/AppIndicator.py:8: PyGIWarning: AppIndicator3 was imported without specifying a version first. Use gi.require_version('AppIndicator3', '0.1') before import to ensure that the right version gets loaded.
  from gi.repository import AppIndicator3 as girAppIndicator
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.StatusIcon.StatusIcon'> 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.Menu.Menu'> 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.ExitItem.ExitItem'> 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.AuthAgent.AuthAgent'> 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.TransferService.TransferService'> 
_________
get_interface_version (/usr/lib/python3/dist-packages/blueman/bluez/obex/Base.py:20)
Detected BlueZ integrated obexd 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.StandardItems.StandardItems'> 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.ShowConnected.ShowConnected'> 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.DBusService.DBusService'> 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.NMPANSupport.NMPANSupport'> 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
Not loading PPPSupport because it's conflict has higher priority 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.NMDUNSupport.NMDUNSupport'> 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
Not loading DhcpClient because it's conflict has higher priority 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.GameControllerWakelock.GameControllerWakelock'> 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.PowerManager.PowerManager'> 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.Networking.Networking'> 
_________
load_nap_settings (/usr/lib/python3/dist-packages/blueman/plugins/applet/Networking.py:36)
Loading NAP settings 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.SerialManager.SerialManager'> 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.AppIndicator.AppIndicator'> 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.Headset.Headset'> 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.DiscvManager.DiscvManager'> 
_________
update_menuitems (/usr/lib/python3/dist-packages/blueman/plugins/applet/DiscvManager.py:123)
warning: Adapter is None 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.KillSwitch.KillSwitch'> 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.RecentConns.RecentConns'> 
_________
io_event (/usr/lib/python3/dist-packages/blueman/plugins/applet/KillSwitch.py:71)
killswitch registered 0 
_________
io_event (/usr/lib/python3/dist-packages/blueman/plugins/applet/KillSwitch.py:71)
State: True 
_________
UpdatePowerState (/usr/lib/python3/dist-packages/blueman/plugins/applet/PowerManager.py:159)
off False 
foff False 
on True 
current state True 
new state True 
_________
io_event (/usr/lib/python3/dist-packages/blueman/plugins/applet/KillSwitch.py:71)
killswitch registered 3 
_________
io_event (/usr/lib/python3/dist-packages/blueman/plugins/applet/KillSwitch.py:71)
State: True 
_________
UpdatePowerState (/usr/lib/python3/dist-packages/blueman/plugins/applet/PowerManager.py:159)
off False 
foff False 
on True 
current state True 
new state True 
_________
_on_obex_owner_changed (/usr/lib/python3/dist-packages/blueman/plugins/applet/TransferService.py:172)
obex owner changed: :1.93 
_________
on_dbus_name_owner_change (/usr/bin/blueman-applet:95)
org.bluez owner changed to :1.183 
_________
register_agent (/usr/lib/python3/dist-packages/blueman/plugins/applet/AuthAgent.py:63)
Registering agent 
_________
set_nap (/usr/lib/python3/dist-packages/blueman/plugins/applet/Networking.py:65)
set nap False 
_________
__init__ (/usr/lib/python3/dist-packages/blueman/main/Device.py:26)
caching initial properties 
_________
initialize (/usr/lib/python3/dist-packages/blueman/plugins/applet/RecentConns.py:144)
rebuilding menu 
_________
enumerate_connections (/usr/lib/python3/dist-packages/blueman/plugins/applet/ShowConnected.py:53)
Found 0 existing connections 
_________
on_registered (/usr/lib/python3/dist-packages/blueman/bluez/obex/AgentManager.py:18)
/org/blueman/obex_agent 
_________
RequestPowerState (/usr/lib/python3/dist-packages/blueman/plugins/applet/PowerManager.py:131)
Requesting True 
_________
on_power_state_change_requested (/usr/lib/python3/dist-packages/blueman/plugins/applet/KillSwitch.py:118)
True 
_________
UpdatePowerState (/usr/lib/python3/dist-packages/blueman/plugins/applet/PowerManager.py:159)
off False 
foff False 
on True 
current state True 
new state True 
_________
check (/usr/lib/python3/dist-packages/blueman/plugins/applet/PowerManager.py:119)
callbacks done 
_________
set_adapter_state (/usr/lib/python3/dist-packages/blueman/plugins/applet/PowerManager.py:90)
True 
```
-------------

hciconfig -a
hci0:	Type: BR/EDR  Bus: USB
	BD Address: 00:1F:E1:E4:D0:6E  ACL MTU: 1017:8  SCO MTU: 64:8
	UP RUNNING PSCAN ISCAN 
	RX bytes:3867265 acl:6344 sco:0 events:419 errors:0
	TX bytes:6643 acl:455 sco:0 commands:39 errors:0
	Features: 0xff 0xff 0x8f 0xfe 0x9b 0xf9 0x00 0x80
	Packet type: DM1 DM3 DM5 DH1 DH3 DH5 HV1 HV2 HV3 
	Link policy: RSWITCH HOLD SNIFF PARK 
	Link mode: SLAVE ACCEPT 
	Name: 'th-ThinkPad-X300'
	Class: 0x1c010c
	Service Classes: Rendering, Capturing, Object Transfer
	Device Class: Computer, Laptop
	HCI Version: 2.0 (0x3)  Revision: 0x212b
	LMP Version: 2.0 (0x3)  Subversion: 0x41d3
	Manufacturer: Broadcom Corporation (15)

<!-- 

ℹ Please use some service like [Pastebin](http://pastebin.com/) or [GitHub Gist](https://gist.github.com/) to post your logs and keep the thread clean and readable. Make sure to describe the exact steps you took when producing it.

->

**Comment by infirit** [comment]
This means hat blueman is not handling the incomming file transfer but gnome-bluetooth is. Upon receiving a file blueman-applet will show it on the console and print some debug info. Most likely you do not see this so below is the start of a transfer I did just now. Yours may look different as you are on a different version.

```
blueman-applet 15.24.12 INFO     Manager:64 _on_object_added: /org/bluez/obex/server/session1
blueman-applet 15.24.12 INFO     Manager:56 _on_object_added: /org/bluez/obex/server/session1/transfer0
blueman-applet 15.24.12 DEBUG    Transfer:25 do_g_properties_changed: /org/bluez/obex/server/session1/transfer0 Status queued
blueman-applet 15.24.12 DEBUG    Transfer:25 do_g_properties_changed: /org/bluez/obex/server/session1/transfer0 Session /org/bluez/obex/server/session1
blueman-applet 15.24.12 DEBUG    Transfer:25 do_g_properties_changed: /org/bluez/obex/server/session1/transfer0 Transferred 0
```

**Comment by insomniacno1** [comment]
Okay, not being an expert here, but what is the solution to this?

**Comment by infirit** [comment]
Do not run gnome-bluetooth and blueman at the same time. Either disable gnome-bluetooth in xfce session 's autostart application or uninstall gnome-bluetooth completely.

**Comment by insomniacno1** [comment]
Okay, the strange part is that this behaviour only started this year around februar - That is when I noticed a file I sent by bluetooth to my computer, disapeared - later I found it in /home/user/.cache/obexd/ . 

I got the Local services in blueman-applet and blueman-manager set to Downloads and it worked for a while, until it didn't. Now it was starting to annoy me and I went looking on the net for a solution which is described in first post - which worked, until it didn't. 

Later I noticed it happen just after an update. Before that, files sent by Bluetooth went into the folder I had set in Local services, which was Public or Pictures or Downloads. 

I never saw the issue on Peach OSI 14.04 LTS and never had it in 2018 on 16.04 LTS.

And I checked gnome-bluetooth is installed(from distributor), but not running.

So are you sure about your diagnose or could it be something else, like a bug in blueman? 
As a general rule I don't uninstall something that other things might rely on.

After all blueman-applet runs on startup and in Local services it is set to Downloads(or whatever folder I have chosen) so according to blueman settings the files received should go to Downloads - but they don't, they go to /home/user/.cache/obexd/

I'm sorry but for me, not being a programmer but troubleshooter and since this issue only started in february, that looks like an programming error in either blueman or any of the other parts it relies on - not something(gnome-bluetooth) that has been there forever. 

So are there any part of this that just got an update and might cause this behaviour or could it be in the kernel itself? 

I just got updated from 4.4.0-151-generic to 4.4.0-154-generic - and in between the fix in the first post and the update, the files went to right place - Can't say that it happen with all updates, but the problem go back to february and there has been alot of updates in between.

With kind regards

Jan

**Comment by cschramm** [comment]
As @infirit said, blueman and gnome-bluetooth compete to handle incoming transfers and it seems like gnome-bluetooth wins. You could check by looking at blueman's output, as @infirit described. If you tried a transfer during in the above log, it did not get handled by blueman.

**Comment by insomniacno1** [comment]
@infirit could you give me the command you used to make the log as it do not show in your answer.

Thank you

**Comment by insomniacno1** [comment]
Did this thread die?

**Comment by cschramm** [comment]
Yes, you did not provide further information or questions. The relevant information is in https://github.com/blueman-project/blueman/issues/1069#issuecomment-508926407. @infirit most probably ran `blueman-applet --loglevel debug` and is on newer version but you should see similar output regarding the file transfer like I said in https://github.com/blueman-project/blueman/issues/1069#issuecomment-509003853.

**Comment by insomniacno1** [comment]
Actually I did ask @infirit  one more time to provide the command he used, so I could try.
And I told you that I would love to try the command but was waiting for @infirit 

Both of these lasted a few minutes to some hours and got deleted from this thread - don't ask me why or how, but they were there when I closed the page.

**Comment by insomniacno1** [comment]
So I ran the command but didn't get the same as @infirit so don't think that was the command he used

user@user:~$ blueman-applet --loglevel debug
blueman-applet version 2.0.4 starting
Stale PID, overwriting
_________
Load (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:60)
['StandardItems', 'PPPSupport', 'ShowConnected', 'NetUsage', 'Menu', 'RecentConns', 'KillSwitch', 'Networking', 'AuthAgent', 'ExitItem', 'NMDUNSupport', 'DBusService', 'PowerManager', 'StatusIcon', 'SerialManager', 'AppIndicator', 'DhcpClient', 'DiscvManager', 'NMPANSupport', 'Headset', 'TransferService', 'GameControllerWakelock'] 
_________
get_interface_version (/usr/lib/python3/dist-packages/blueman/bluez/BlueZInterface.py:13)
Detected BlueZ 5 
/usr/lib/python3/dist-packages/blueman/plugins/applet/AppIndicator.py:8: PyGIWarning: AppIndicator3 was imported without specifying a version first. Use gi.require_version('AppIndicator3', '0.1') before import to ensure that the right version gets loaded.
  from gi.repository import AppIndicator3 as girAppIndicator
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.StatusIcon.StatusIcon'> 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.AuthAgent.AuthAgent'> 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.GameControllerWakelock.GameControllerWakelock'> 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.Networking.Networking'> 
_________
load_nap_settings (/usr/lib/python3/dist-packages/blueman/plugins/applet/Networking.py:36)
Loading NAP settings 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.ShowConnected.ShowConnected'> 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.Menu.Menu'> 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.AppIndicator.AppIndicator'> 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.Headset.Headset'> 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.DBusService.DBusService'> 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.NMPANSupport.NMPANSupport'> 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.NMDUNSupport.NMDUNSupport'> 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.DiscvManager.DiscvManager'> 
_________
update_menuitems (/usr/lib/python3/dist-packages/blueman/plugins/applet/DiscvManager.py:123)
warning: Adapter is None 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.PowerManager.PowerManager'> 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.SerialManager.SerialManager'> 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.StandardItems.StandardItems'> 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.TransferService.TransferService'> 
_________
get_interface_version (/usr/lib/python3/dist-packages/blueman/bluez/obex/Base.py:20)
Detected BlueZ integrated obexd 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
Not loading DhcpClient because it's conflict has higher priority 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.RecentConns.RecentConns'> 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.KillSwitch.KillSwitch'> 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.ExitItem.ExitItem'> 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
Not loading PPPSupport because it's conflict has higher priority 
_________
io_event (/usr/lib/python3/dist-packages/blueman/plugins/applet/KillSwitch.py:71)
killswitch registered 0 
_________
io_event (/usr/lib/python3/dist-packages/blueman/plugins/applet/KillSwitch.py:71)
State: True 
_________
UpdatePowerState (/usr/lib/python3/dist-packages/blueman/plugins/applet/PowerManager.py:159)
off False 
foff False 
on True 
current state True 
new state True 
_________
io_event (/usr/lib/python3/dist-packages/blueman/plugins/applet/KillSwitch.py:71)
killswitch registered 1 
_________
io_event (/usr/lib/python3/dist-packages/blueman/plugins/applet/KillSwitch.py:71)
State: True 
_________
UpdatePowerState (/usr/lib/python3/dist-packages/blueman/plugins/applet/PowerManager.py:159)
off False 
foff False 
on True 
current state True 
new state True 
_________
_on_obex_owner_changed (/usr/lib/python3/dist-packages/blueman/plugins/applet/TransferService.py:172)
obex owner changed: :1.93 
_________
on_registered (/usr/lib/python3/dist-packages/blueman/bluez/obex/AgentManager.py:18)
/org/blueman/obex_agent 
_________
on_dbus_name_owner_change (/usr/bin/blueman-applet:95)
org.bluez owner changed to :1.2 
_________
register_agent (/usr/lib/python3/dist-packages/blueman/plugins/applet/AuthAgent.py:63)
Registering agent 
_________
set_nap (/usr/lib/python3/dist-packages/blueman/plugins/applet/Networking.py:65)
set nap False 
_________
__init__ (/usr/lib/python3/dist-packages/blueman/main/Device.py:26)
caching initial properties 
_________
initialize (/usr/lib/python3/dist-packages/blueman/plugins/applet/RecentConns.py:144)
rebuilding menu 
_________
enumerate_connections (/usr/lib/python3/dist-packages/blueman/plugins/applet/ShowConnected.py:53)
Found 0 existing connections 
_________
RequestPowerState (/usr/lib/python3/dist-packages/blueman/plugins/applet/PowerManager.py:131)
Requesting True 
_________
on_power_state_change_requested (/usr/lib/python3/dist-packages/blueman/plugins/applet/KillSwitch.py:118)
True 
_________
UpdatePowerState (/usr/lib/python3/dist-packages/blueman/plugins/applet/PowerManager.py:159)
off False 
foff False 
on True 
current state True 
new state True 
_________
check (/usr/lib/python3/dist-packages/blueman/plugins/applet/PowerManager.py:119)
callbacks done 
_________
set_adapter_state (/usr/lib/python3/dist-packages/blueman/plugins/applet/PowerManager.py:90)
True 

Sending a file ad this to the terminal

_________
__init__ (/usr/lib/python3/dist-packages/blueman/main/Device.py:26)
caching initial properties 
_________
__del__ (/usr/lib/python3/dist-packages/blueman/main/Device.py:72)
deleting device /org/bluez/hci0/dev_70_3A_51_96_27_4B 
_________
Destroy (/usr/lib/python3/dist-packages/blueman/main/Device.py:94)
invalidating device /org/bluez/hci0/dev_70_3A_51_96_27_4B 
_________
_on_transfer_started (/usr/lib/python3/dist-packages/blueman/bluez/obex/Manager.py:61)
/org/bluez/obex/server/session15/transfer14 
_________
AuthorizePush (/usr/lib/python3/dist-packages/blueman/bluez/obex/Agent.py:42)
/org/blueman/obex_agent /org/bluez/obex/server/session15/transfer14 
_________
__init__ (/usr/lib/python3/dist-packages/blueman/main/Device.py:26)
caching initial properties 
_________
_on_action (/usr/lib/python3/dist-packages/blueman/plugins/applet/TransferService.py:48)
accept 
_________
reply (/usr/lib/python3/dist-packages/blueman/bluez/obex/Agent.py:66)
/org/blueman/obex_agent /home/th/.cache/obexd/Krav til kat og urne.txt 
_________
__del__ (/usr/lib/python3/dist-packages/blueman/main/Device.py:72)
deleting device /org/bluez/hci0/dev_70_3A_51_96_27_4B 
_________
Destroy (/usr/lib/python3/dist-packages/blueman/main/Device.py:94)
invalidating device /org/bluez/hci0/dev_70_3A_51_96_27_4B 
_________
_on_properties_changed (/usr/lib/python3/dist-packages/blueman/bluez/obex/Transfer.py:54)
/org/bluez/obex/server/session15/transfer14 Status active 
_________
_on_properties_changed (/usr/lib/python3/dist-packages/blueman/bluez/obex/Transfer.py:54)
/org/bluez/obex/server/session15/transfer14 Transferred 587 
_________
_on_properties_changed (/usr/lib/python3/dist-packages/blueman/bluez/obex/Transfer.py:54)
/org/bluez/obex/server/session15/transfer14 Status active 
_________
_on_properties_changed (/usr/lib/python3/dist-packages/blueman/bluez/obex/Transfer.py:54)
/org/bluez/obex/server/session15/transfer14 Transferred 587 
_________
_on_properties_changed (/usr/lib/python3/dist-packages/blueman/bluez/obex/Transfer.py:54)
/org/bluez/obex/server/session15/transfer14 Transferred 1174 
_________
_on_properties_changed (/usr/lib/python3/dist-packages/blueman/bluez/obex/Transfer.py:54)
/org/bluez/obex/server/session15/transfer14 Transferred 1174 
_________
_on_properties_changed (/usr/lib/python3/dist-packages/blueman/bluez/obex/Transfer.py:54)
/org/bluez/obex/server/session15/transfer14 Transferred 1761 
_________
_on_properties_changed (/usr/lib/python3/dist-packages/blueman/bluez/obex/Transfer.py:54)
/org/bluez/obex/server/session15/transfer14 Transferred 1761 
_________
_on_properties_changed (/usr/lib/python3/dist-packages/blueman/bluez/obex/Transfer.py:54)
/org/bluez/obex/server/session15/transfer14 Transferred 2348 
_________
_on_properties_changed (/usr/lib/python3/dist-packages/blueman/bluez/obex/Transfer.py:54)
/org/bluez/obex/server/session15/transfer14 Transferred 2348 
_________
_on_properties_changed (/usr/lib/python3/dist-packages/blueman/bluez/obex/Transfer.py:54)
/org/bluez/obex/server/session15/transfer14 Transferred 2935 
_________
_on_properties_changed (/usr/lib/python3/dist-packages/blueman/bluez/obex/Transfer.py:54)
/org/bluez/obex/server/session15/transfer14 Transferred 2935 
_________
_on_properties_changed (/usr/lib/python3/dist-packages/blueman/bluez/obex/Transfer.py:54)
/org/bluez/obex/server/session15/transfer14 Transferred 3522 
_________
_on_properties_changed (/usr/lib/python3/dist-packages/blueman/bluez/obex/Transfer.py:54)
/org/bluez/obex/server/session15/transfer14 Transferred 3522 
_________
_on_properties_changed (/usr/lib/python3/dist-packages/blueman/bluez/obex/Transfer.py:54)
/org/bluez/obex/server/session15/transfer14 Transferred 4109 
_________
_on_properties_changed (/usr/lib/python3/dist-packages/blueman/bluez/obex/Transfer.py:54)
/org/bluez/obex/server/session15/transfer14 Transferred 4109 
_________
_on_properties_changed (/usr/lib/python3/dist-packages/blueman/bluez/obex/Transfer.py:54)
/org/bluez/obex/server/session15/transfer14 Transferred 4696 
_________
_on_properties_changed (/usr/lib/python3/dist-packages/blueman/bluez/obex/Transfer.py:54)
/org/bluez/obex/server/session15/transfer14 Transferred 4696 
_________
_on_properties_changed (/usr/lib/python3/dist-packages/blueman/bluez/obex/Transfer.py:54)
/org/bluez/obex/server/session15/transfer14 Transferred 5283 
_________
_on_properties_changed (/usr/lib/python3/dist-packages/blueman/bluez/obex/Transfer.py:54)
/org/bluez/obex/server/session15/transfer14 Transferred 5283 
_________
_on_properties_changed (/usr/lib/python3/dist-packages/blueman/bluez/obex/Transfer.py:54)
/org/bluez/obex/server/session15/transfer14 Transferred 5870 
_________
_on_properties_changed (/usr/lib/python3/dist-packages/blueman/bluez/obex/Transfer.py:54)
/org/bluez/obex/server/session15/transfer14 Transferred 5870 
_________
_on_properties_changed (/usr/lib/python3/dist-packages/blueman/bluez/obex/Transfer.py:54)
/org/bluez/obex/server/session15/transfer14 Transferred 6457 
_________
_on_properties_changed (/usr/lib/python3/dist-packages/blueman/bluez/obex/Transfer.py:54)
/org/bluez/obex/server/session15/transfer14 Transferred 6457 
_________
_on_properties_changed (/usr/lib/python3/dist-packages/blueman/bluez/obex/Transfer.py:54)
/org/bluez/obex/server/session15/transfer14 Transferred 6518 
_________
_on_properties_changed (/usr/lib/python3/dist-packages/blueman/bluez/obex/Transfer.py:54)
/org/bluez/obex/server/session15/transfer14 Transferred 6518 
_________
_on_properties_changed (/usr/lib/python3/dist-packages/blueman/bluez/obex/Transfer.py:54)
/org/bluez/obex/server/session15/transfer14 Status complete 
_________
_on_transfer_completed (/usr/lib/python3/dist-packages/blueman/bluez/obex/Manager.py:65)
/org/bluez/obex/server/session15/transfer14 True 
_________
_on_transfer_completed (/usr/lib/python3/dist-packages/blueman/plugins/applet/TransferService.py:201)
Destination file exists, renaming to: 20190723114445_Krav til kat og urne.txt 

(blueman-applet:17165): Gdk-CRITICAL **: gdk_window_get_origin: assertion 'GDK_IS_WINDOW (window)' failed
adding action
_________
_on_properties_changed (/usr/lib/python3/dist-packages/blueman/bluez/obex/Transfer.py:54)
/org/bluez/obex/server/session15/transfer14 Status complete 
_________
_on_session_removed (/usr/lib/python3/dist-packages/blueman/bluez/obex/Manager.py:57)
/org/bluez/obex/server/session15 
_________
__init__ (/usr/lib/python3/dist-packages/blueman/main/Device.py:26)
caching initial properties 
_________
__del__ (/usr/lib/python3/dist-packages/blueman/main/Device.py:72)
deleting device /org/bluez/hci0/dev_70_3A_51_96_27_4B 
_________
Destroy (/usr/lib/python3/dist-packages/blueman/main/Device.py:94)
invalidating device /org/bluez/hci0/dev_70_3A_51_96_27_4B 
_________
__init__ (/usr/lib/python3/dist-packages/blueman/main/Device.py:26)
caching initial properties 
_________
__del__ (/usr/lib/python3/dist-packages/blueman/main/Device.py:72)
deleting device /org/bluez/hci0/dev_70_3A_51_96_27_4B 
_________
Destroy (/usr/lib/python3/dist-packages/blueman/main/Device.py:94)
invalidating device /org/bluez/hci0/dev_70_3A_51_96_27_4B

**Comment by insomniacno1** [comment]
For me it looks like its ONLY blueman that handles this and not gnome-bluetooth

**Comment by infirit** [comment]
@insomniacno1 that is exactly what we need to see what is going on. The output is different as the current version 2.1 has diverged a lot and that is what I am using.

From a terminal what is the output of ``gsettings get org.blueman.transfer shared-path``.  The output will show you what path blueman will use to move the incoming file to. If this is not what you expect run ``blueman-services`` and configure it in the "Transfer" section.

**Comment by insomniacno1** [comment]
@infirit I did everything as you asked and it is now set to Downloads folder and the files go there. But as I explained before, I did this before and just after any update it was back to the obexd folder eventhough it was set to Downloads in blueman-services. I will leave it like this and then get back to you after next update, if it changes again.

**Comment by infirit** [comment]
You did not share what the gsettings command returned, an empty string, a path to your cache dir or something completely different? If you still have the output post it otherwise include this when you report back :smile: .

You can monitor changes to this setting with ``gsettings monitor org.blueman.transfer shared-path``.

**Comment by insomniacno1** [comment]
@infirit gsettings get org.blueman.transfer shared-path shows whatever is configured in blueman-services whether that is obexd or Downloads - right now '/home/th/Downloads'

gsettings monitor org.blueman.transfer shared-path does nothing, it doesn't even go back to prompt, it just freeze the cursor so that Ctrl + C is only option.

I updated yesterday and it seems to be stable for now, so I will say its solved.

Thank you for the help :-)

With kind regards

Jan

**Comment by iago-pssjd** [comment]
Hi @cschramm @infirit 

I am having a similar issue. If I run `gsettings monitor org.blueman.transfer shared-path` it does not return anything (it hangs up), and if I run `gsettings get org.blueman.transfer shared-path`, if I do it as a user, then I get `/home/user/Downloads` (as I configure through GUI), but if I run it as root I get `''`.  Can I configure this the Transfer Incoming folder through command line? How shouls I do it?

Thanks!

**Comment by cschramm** [comment]
@iago-pssjd: If `gsettings get org.blueman.transfer shared-path` gives you the expected path, it's already configured. If it does not work, please open a new issue with the output from `blueman-applet --loglevel debug` when receiving a file.

**Comment by iago-pssjd** [comment]
That's it, my version is exactly 2.0.4
